### ISSUE TITLE
Carity in UI on when a node creates a resource within the engine

### ISSUE THREAD

**Issue by LaylBongers** [OurMachinery/themachinery-public#661]
It tends to be somewhat unclear to me when something is created by a graph. Especially in the case of shaders and shader instances, where that difference is important. As well it's somewhat unclear how inherited graphs get affected by this. Do specific nodes 'compile' a group of nodes together into a resource? Maybe display some information on nodes' headers on what kind of resource they create if that's the case?

**Comment by tobias-ourmachinery** [comment]
I agree. We'll work on improving clarity around this. I think we probably want to both have a visual indication on the nodes that resulted in a new buffer being created in The Truth, as well as some way to get a resource breakdown / summary for a creation graph asset.
